packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -246,15 +270,24 @@ export class RedirectsMiddleware extends MiddlewareBase {
           // Save the matched query string (if found) into the redirect object
           redirect.matchedQueryString = matchedQueryString || '';
 
-          // Return the redirect if the URL path or any query string permutation matches the pattern
-          return (
-            (regexParser(redirect.pattern).test(targetURL) ||
+          const matchedPatterResult =
+            !!(
+              regexParser(redirect.pattern).test(targetURL) ||
               regexParser(redirect.pattern).test(`/${req.nextUrl.locale}${targetURL}`) ||
-              matchedQueryString) &&
-            (redirect.locale ? redirect.locale.toLowerCase() === locale.toLowerCase() : true)
-          );
+              matchedQueryString
+            ) && (redirect.locale ? redirect.locale.toLowerCase() === locale.toLowerCase() : true);
+
+          // Save cache the result for the current pattern
+          this.cache.setCacheValue(chachedPatternResultKey, matchedPatterResult);
```

Comment:
I'm thinking, if all these calculations will ever be executed if you have a [cache](https://github.com/Sitecore/jss/blob/bugifx/SXA-7834-improvement-perfomence-redirects/packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts#L206)
I think not really, so you don't need to set additional cache for patterns, since anyway it's based on [cacheKey](https://github.com/Sitecore/jss/blob/bugifx/SXA-7834-improvement-perfomence-redirects/packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts#L217)
If `cacheKey` cache already present you will not traverse all the `fetchedRedirects` again